lib/mox/server.ex

```diff
@@ -0,0 +1,71 @@
+defmodule Mox.Server do
+  @moduledoc false
+
+  use GenServer
+
+  # Public API
+
+  def start_link(_options) do
+    GenServer.start_link(__MODULE__, %{}, name: __MODULE__)
+  end
+
+  def add_expectation(key, {n, calls}) do
+    GenServer.call(__MODULE__, {:add_expectation, key, {n, calls}})
+  end
+
+  def fetch_expectation_for_dispatch(key) do
+    case GenServer.call(__MODULE__, {:fetch_expectation_for_dispatch, key}) do
+      nil -> :error
+      expectation -> expectation
+    end
+  end
+
+  def fetch_expectation_for_verify(key) do
+    GenServer.call(__MODULE__, {:fetch_expectation_for_verify, key})
+  end
+
+  def keys(owner_pid) do
+    GenServer.call(__MODULE__, {:keys, owner_pid})
+  end
+
+  # Callbacks
+
+  def handle_call({:add_expectation, key, {n, calls}}, _from, state) do
+    new_state = case state[key] do
```

Comment:
This can be implemented as a `Map.update/4` without the case.

Reply:
sure